chore: update GitHub actions workflows

tylervz opened the pull request:
Update the GitHub actions workflows to use the latest version of multiple actions, including `actions/checkout`,  `actions/setup-java`, `actions/cache`, and `release-drafter/release-drafter`:

https://github.com/actions/checkout/tags
https://github.com/actions/setup-java/tags
https://github.com/actions/cache?tab=readme-ov-file#v4
https://github.com/release-drafter/release-drafter/tags

puneetbehl committed:
New Baseline to Java 17 (#1765)

puneetbehl committed:
Merge branch '8.0.x' into 9.0.x

puneetbehl committed:
Merge branch '8.0.x' into 9.0.x

tylervz committed:
chore: update GH actions workflows

tylervz commented on .github/workflows/gradle.yml:
```diff
@@ -13,44 +13,54 @@ jobs:
       - name: Run Build
         id: build
-        uses: gradle/gradle-build-action@v2
+        run: ./gradlew build
```
The `gradle/actions/setup-gradle` action replaces `gradle/gradle-build-action`.
Reference:
https://github.com/gradle/actions/tree/af6e5767241073b8ab20cc8cf7e95ec63fadc160/setup-gradle#the-setup-gradle-action

tylervz commented on .github/workflows/release-notes.yml:
```diff
@@ -24,7 +24,7 @@ jobs:
         run: echo "value=${GITHUB_REF:11}" >> $GITHUB_OUTPUT
       # If it has release drafter:
-      - uses: release-drafter/release-drafter@v5
+      - uses: release-drafter/release-drafter@v6
```
There are no breaking changes in this release:
https://github.com/release-drafter/release-drafter/releases/tag/v6.0.0

tylervz commented:
I did not upgrade the ["Build and install groovy (no docs)" step in `groovy-joint-workflow.yml`](https://github.com/grails/grails-data-mapping/pull/1806/files#diff-cb75eba9bba0d6665705e7a07feb9286848cc414d3bcd32aadcbe925c5d96c3cL97-R111), because I did not see a suitable replacement using `gradle/actions/setup-gradle@v3`.

If I replaced it with `gradle/actions/setup-gradle@v3` like so:
```yaml
- name: Setup Gradle
uses: gradle/actions/setup-gradle@v3
with:
  build-scan-publish: true
  build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
  build-scan-terms-of-use-agree: "yes"
# Use the Gradle wrapper in the ../groovy directory
- name: Build and install groovy (no docs)
  run: |
    cd ../groovy
    ./gradlew install \
      -x groovydoc \
      -x javadoc \
      -x javadocAll \
      -x groovydocAll \
      -x asciidoc \
      -x docGDK
  env:
    GRADLE_SCANS_ACCEPT: yes
    GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
Then the `gradle/actions/setup-gradle@v3` step in the example above is useless since it is configuring Gradle in the root directory for optimal execution rather than the Gradle wrapper in `../groovy` that is actually going to get used.

I do not see any inputs/parameters for `gradle/actions/setup-gradle@v3` similar to `build-root-directory` to have it configure a Gradle wrapper in a directory other than the root directory.
https://github.com/gradle/actions/blob/main/setup-gradle/action.yml

Update: perhaps we should update the step to be the same as it is in `grails-core`?
https://github.com/grails/grails-core/blob/6.2.x/.github/workflows/groovy-joint-workflow.yml#L103-L119

jamesfredley committed:
Merge branch '9.0.x' into jamesfredley/9.0.x-changes

# Conflicts:
#	gradle.properties

jamesfredley committed:
javax to jakarta, groovy 4, spring framework 6 and spring boot 3 migrations (#1810)

* javax to jakarta, groovy 4, spring framework 6 and spring boot 3 migrations
---------

Co-authored-by: Mattias Reichel <mattias.reichel@gmail.com>
Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: puneetbehl <behlp@objectcomputing.com>
Co-authored-by: Puneet Behl <behlp@unityfoundation.io>
Co-authored-by: Guillermo Calvo <guillermocalvo@gmail.com>
Co-authored-by: jhron <jhron@users.noreply.github.com>
Co-authored-by: micronaut-build <puneetbehl@users.noreply.github.com>
Co-authored-by: Guillermo Calvo <guillermo@guillermo.dev>

renovate[bot] committed:
fix(deps): update dependency org.javassist:javassist to v3.30.2-ga (#1817)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency commons-validator:commons-validator to v1.9.0 (#1805)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update plugin com.gradle.enterprise to v3.18 (#1804)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update spring core to v6.1.12 (#1803)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update spring core to v6.1.13 (#1820)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency gradle to v8.10.1 (#1818)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update plugin com.gradle.enterprise to v3.18.1 (#1819)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update groovy monorepo to v4.0.23 (#1821)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jamesfredley committed:
Move all versions to gradle.properties, update dependencies to latest compatible version and remove unused (#1825)

renovate[bot] committed:
chore(deps): update dependency gradle to v8.10.2 (#1826)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency com.h2database:h2 to v2.3.232

codeconsole committed:
get rid of commons-lang

codeconsole committed:
Merge pull request #1814 from grails/renovate/h2version

fix(deps): update dependency com.h2database:h2 to v2.3.232

codeconsole committed:
el libraries are provided by container

jamesfredley committed:
compileOnlyApi jakarta.el (#1829)

jamesfredley committed:
Update to grails-async 6.0.0-SNAPSHOT, spring 6.1.14 and us hibernate-core-jakarta (#1831)

jamesfredley committed:
restore required glassfish el implementation since it broke all test applications in gorm-hibernate5 (#1832)

jamesfredley committed:
add grails-bom to resolve dependency versions

jamesfredley committed:
use Groovy 4.0.24-SNAPSHOT

jamesfredley committed:
restore domain inheritance tests

jamesfredley committed:
adjust Domain examples

jamesfredley committed:
Merge pull request #1834 from grails/restore-domain-inheritance

Restore domain inheritance tests

jamesfredley committed:
remove comments

jamesfredley committed:
trigger hibernate6 repository build github workflow

jamesfredley committed:
skip test for hibernate6 in addition to hibernate5

jamesfredley committed:
Merge pull request #1835 from grails/test-cleanup

test cleanup - fix hibernate6 tests

jdaugherty committed:
Fix #1155 - validate(args) should be consistent with validate()

jdaugherty committed:
Cleanup & fix annotation reading for classpath scanning

jdaugherty committed:
Merge pull request #1836 from jdaugherty/9.0.x

Cleanup & fix annotation reading for classpath scanning

jdaugherty committed:
Revert classpath scanning change, use the same strategy as grails-core

jdaugherty committed:
Merge pull request #1837 from jdaugherty/9.0.x

Revert classpath scanning change, use the same strategy as grails-core

jdaugherty committed:
Merge branch '9.0.x' into feature/9.0.x/1155-skip-validate

jdaugherty committed:
Merge pull request #1838 from jdaugherty/feature/9.0.x/1155-skip-validate

codeconsole committed:
Remove api dependency until groovy version stabilizes

codeconsole committed:
Merge pull request #1839 from codeconsole/9.0.x-no-groovy-api

Remove api dependency until groovy version stabilizes

codeconsole committed:
remove groovy transitive dependency until groovy version stabilizes

codeconsole committed:
Merge pull request #1840 from codeconsole/9.0.x-no-groovy-api

remove groovy transitive dependency until groovy version stabilizes

codeconsole committed:
dependency cleanup

codeconsole committed:
Merge pull request #1841 from codeconsole/9.0.x-bom-cleanup

dependency cleanup

codeconsole committed:
Merge branch '8.1.x' into 9.0.x

codeconsole committed:
jdk 17

renovate[bot] committed:
Update groovy monorepo to v4.0.24

jdaugherty committed:
Merge pull request #1842 from grails/renovate/groovy-monorepo

Update groovy monorepo to v4.0.24

jdaugherty committed:
Updates since groovy 4.0.24 is released

jdaugherty committed:
Merge pull request #1843 from jdaugherty/9.0.x

Updates since groovy 4.0.24 is released

jdaugherty committed:
Add back logic for groovy snapshot support

jdaugherty committed:
Merge pull request #1844 from jdaugherty/9.0.x

Add back logic for groovy snapshot support

renovate[bot] committed:
Update dependency gradle to v8.11

codeconsole committed:
Merge pull request #1846 from grails/renovate/gradle-8.x

Update dependency gradle to v8.11

jamesfredley commented:
@tylervz after https://github.com/grails/grails-data-mapping/pull/1849 has been merged, can you rebase this against 9.0.x?

tylervz commented:
@jamesfredley yes, I can do that :)

matrei committed:
ci: Update `groovy-joint-workflow` to use `mavenLocal()` (#1849)

Also update to develocity gradle plugin and related change to DEVELOCITY_* env vars.

tylervz commented:
I've rebased my branch on `9.0.x`. I'm going to close this pull request and open a new one requesting to be merged into `9.0.x`.
